odoo/models.py

```diff
@@ -1832,13 +1832,18 @@ def _compute_display_name(self):
             record.display_name = names.get(record.id, False)
 
     def name_get(self):
-        """ name_get() -> [(id, name), ...]
+        """Returns a textual representation for the records in ``self``, with
+        one item output per input record, in the same order.
 
-        Returns a textual representation for the records in ``self``.
-        By default this is the value of the ``display_name`` field.
+        .. warning::
+
+            Although :meth:`~.name_get` can use context data for richer
+            contextual formatting, as it is the default implementation for
+            :attr:`~.display_name` it is important that it resets to the
+            "default" behaviour if the context keys are empty / missing.
```

Comment:
Wait, now this becomes confusing.  The contextual formatting is a feature of model `res.partner`, and that's why `_compute_display_name` is overridden on _that_ model to avoid fancy names.  If this documentation is valid for _all_ models, then the default implementation of `_compute_display_name` should take it into account.  It currently doesn't, and it does not matter that much, since `display_name` is not stored by default.

Reply:
To be complete, my initial suggestion was to document/comment `name_get()` on model `res.partner` about the convention taken by `display_name`.

Reply:
> The contextual formatting is a feature of model `res.partner`.

While res.partner makes significant use of it, and it is singularly problematic for it (due to the field being stored), it's nowhere close to the only model/module with a context-dependent `name_get` e.g. `account.move`, `hr.department`, `hr_timesheet`, `documents`, `sale_planning`, `sale_renting`, ...

As you note it's way less problematic for the rest as they're not stored, but I think the warning is valid in general regardless.

Reply:
Ok fine, your argument is relevant and reasonable.

Reply:
Something to consider for the future if we ever replace the ad-hoc grepping by semgrep would be a lint like `orm` for this sort of unadvisable or tricky uses e.g. we could have a warning iff the context is used in a `name_get` or `compute` method.